tests/cli_test.py

```diff
@@ -125,12 +127,267 @@ def cleos_sign_test():
+        processCleosCommand(cmd.split())
+
+        accounts = ["eosio.token", "alice", "bob"]
+        for account in accounts:
+            cmd = "./programs/cleos/cleos get account -j " + account
+            outs, errs = processCleosCommand(cmd.split())
+            str = '"account_name"' + ': ' + '"' + account + '"'
+            if str not in outs.decode('utf-8'):
+                cmd = "./programs/cleos/cleos -u http://127.0.0.1:8888 create account eosio " + account + " " + PUBLIC_KEY
+                outs, errs = processCleosCommand(cmd.split())
+
+        cmd = "./programs/cleos/cleos set contract eosio.token " + os.path.abspath(os.getcwd() + "/../unittests/contracts/eosio.token") + " " + "--abi " + "eosio.token.abi -p eosio.token@active"
+        processCleosCommand(cmd.split())
+        cmd = ['./programs/cleos/cleos', 'push', 'action', 'eosio.token', 'create', '[ "alice", "1000000000.0000 SYS" ]', '-p', 'eosio.token']
+        processCleosCommand(cmd)
+        cmd = ['./programs/cleos/cleos', 'push', 'action', 'eosio.token', 'issue', '[ "alice", "100.0000 SYS", "memo" ]', '-p', 'alice']
+        processCleosCommand(cmd)
+
+        # make a malicious abi file by switching 'from' and 'to' in eosio.token.abi
+        malicious_token_abi_path = os.path.abspath(os.getcwd() + '/../unittests/contracts/eosio.token/malicious.eosio.token.abi')
+        shutil.copyfile(token_abi_path, malicious_token_abi_path)
+        replaces = [["from", "malicious"], ["to", "from"], ["malicious", "to"]]
+        for replace in replaces:
+            with open(malicious_token_abi_path, 'r+') as f:
+                abi = f.read()
+                abi = re.sub(replace[0], replace[1], abi)
+                f.seek(0)
+                f.write(abi)
+                f.truncate()
+
+        # set the malicious abi
+        cmd = "./programs/cleos/cleos set abi eosio.token " + malicious_token_abi_path
+        processCleosCommand(cmd.split())
+
+        # option '--abi-file' makes token still be transferred from alice to bob after setting the malicious abi
+        cmd = ['./programs/cleos/cleos', '-u', 'http://127.0.0.1:8888', '--print-request', '--abi-file', token_abi_file_arg, 'push', 'action', 'eosio.token', 'transfer', '{ "from":"alice", "to":"bob", "quantity":"25.0000 SYS", "memo":"m" }', '-p', 'alice']
+        outs, errs = processCleosCommand(cmd)
+        assert(b'"/v1/chain/get_raw_abi"' not in errs)
+        cmd = "./programs/cleos/cleos get currency balance eosio.token alice SYS"
+        outs, errs = processCleosCommand(cmd.split())
+        assert(outs.strip().decode('utf-8') == "75.0000 SYS")
+
+    finally:
+
+        if pNodeos.pid:
+            os.kill(pNodeos.pid, signal.SIGKILL)
+        shutil.rmtree(data_dir)
+
+        if pWallet.pid:
+            os.kill(pWallet.pid, signal.SIGKILL)
+        subprocess.call(("pkill -9 keosd").split())
+        wallet_file = os.path.expanduser('~') + "/eosio-wallet/" + wallet_name + ".wallet"
```

Comment:
This is deleting wallet files in the user's default wallet directory. Must be changed.

Reply:
Least intrusive is one thing, but this is horrific.  Will be fixed.